### PR TITLE
Convert LinearGradient and RadialGradient to fractional units

### PR DESCRIPTION
**Pull request by Hixie**
Makes it easier to use without a SizeObserver

### CODE REVIEW

**Comment by Hixie** [comment]
@abarth

File: examples/material_gallery/lib/demo/weathers_demo.dart [hunk 1]
```diff
@@ -276,14 +276,15 @@ class GradientNode extends NodeWithSize {
   void paint(Canvas canvas) {
     applyTransformForPivot(canvas);
 
+    Rect rect = Point.origin & size;
     Paint gradientPaint = new Paint()..shader = new LinearGradient(
-      begin: Point.origin,
-      end: new Point(0.0, size.height),
+      begin: const Offset(0.0, 0.0),
```
Comment (abarth) [inline examples/material_gallery/lib/demo/weathers_demo.dart]: Offset.zero

File: packages/flutter/lib/src/painting/box_painter.dart [hunk 2]
```diff
@@ -235,27 +235,41 @@ class BoxShadow {
   String toString() => 'BoxShadow($color, $offset, $blurRadius, $spreadRadius)';
 }
 
+Point _offsetToPoint(Offset offset, Rect rect) {
+  return new Point(rect.left + offset.dx * rect.width, rect.top + offset.dy * rect.height);
+}
```
Comment (abarth) [inline packages/flutter/lib/src/painting/box_painter.dart]: This should be a function on Rect.  File a bug and put a TODO comment here?

File: packages/flutter/test/widget/shader_mask_test.dart [hunk 3]
```diff
@@ -10,12 +10,12 @@ import 'package:test/test.dart';
 
 Shader createShader(Rect bounds) {
   return new LinearGradient(
-      begin: bounds.topLeft,
-      end: bounds.bottomLeft,
+      begin: const Offset(0.0, 0.0),
```
Comment (abarth) [inline packages/flutter/test/widget/shader_mask_test.dart]: Offset.zero

**Comment by abarth** [comment]
LGTM

File: packages/flutter/lib/src/painting/box_painter.dart (hunk 2 above)
Comment (Hixie) [inline packages/flutter/lib/src/painting/box_painter.dart]: https://github.com/flutter/flutter/issues/2318

**Pull request merged**